packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -100,8 +114,14 @@ class TransferEngine implements ITransferEngine {
 
   async transfer(): Promise<void> {
     try {
+      // Bootstrap the providers
```

Comment:
This is more an overall note for what I've seen in the whole project, do we need that many comments? In my opinion, we should only use comments when adding code that is difficult to understand, as Robert C. Martin states https://www.baeldung.com/cs/clean-code-comments. What do you think about this? 😄

Reply:
Good point, we never really had any convention regarding this.

For OSS libraries, it's always a bit hard to draw the line between what's self-explicit & what needs to be explained, as you cannot assume the codebase (and overall) knowledge of the reader, hence the many comments in this kind of orchestration structures. 

If this is something that you like to see change, maybe you could create a contributor documentation page PR regarding the comments convention? This way, every contributor would benefit from it. What do you think?

Reply:
+1, would love to see some suggested standards around this